Adds Grants tab to Give

0xJem opened the pull request:
Implements the Grants interface under Give.

The GrantCard component (copied from ProjectCard) has been implemented using Material UI's Grid component, instead of CSS and manual breakpoints. This should result in more predictable behaviour and less technical debt.

This also includes #1545, which moves all Give components to use the Material UI Grid component. Why?
- Layout, particularly breakpoint behaviour, is written inline in JSX, instead of CSS
- The amount of CSS+JSX code is significantly (order of magnitude?) less, which reduces technical debt and makes it easier to understand
- Less prone to inadvertent failure due to changing CSS

TODO:
- [x] Sign-off from design
- [x] Replace wonky SVG icons
- [ ] Load final grant data (milestones are missing) - might go into a future PR
- [x] Ability to disable individual grants

0xJem committed:
Cleanup of redundant variables

0xJem committed:
Remove redundant types and variables

0xJem committed:
Fix issue with type definition for donationInfo

0xJem committed:
Restore yield image that is incorrect in the component library

brightiron committed:
removing svgs

0xJem committed:
Fix colour of vault icon

0xJem committed:
Merge pull request #1344 from OlympusDAO/brightiron/yieldVaultSvgGive

removing bundled svgs

0xJem committed:
Tab indicator 4px from tab

0xJem committed:
Shift spacing between give tabs and content to 40px

0xJem committed:
Align project title with image

0xJem committed:
Prevent wrapping of give info titles

0xJem committed:
Hopefully a fix for translation issues

0xJem committed:
Merge pull request #1354 from OlympusDAO/tyche-causes-design-feedback

Design feedback on Give causes v2

0xLienid committed:
Vertical spacing of cause card content

0xLienid committed:
Update sOHM goal copy

0xLienid committed:
Confirmation modal text padding fix

0xLienid committed:
Fixed edit and withdrawal modal close calls

0xLienid committed:
Update indication of project name in deposit modal

0xJem committed:
Testing a different approach to translation strings

0xJem committed:
Another approach to translations

0xJem committed:
Trying different approaches to translation

0xJem committed:
Translation fix and update README

0xJem committed:
Linting fixes

0xJem committed:
Constrain give info component on smaller screens

0xJem committed:
Top-align project information on medium breakpoint

0xJem committed:
Constrain project image height

0xJem committed:
Consistent 30px spacing on project card

0xJem committed:
Vertical alignment on mobile

0xJem committed:
30px margin on mobile view

0xJem committed:
Better handling of breakpoints

0xJem committed:
Cleaner className when checking for screen size

0xJem committed:
Fix give info on large screens

0xLienid committed:
Redeem tab-panel view desktop, tablet, mobile

0xLienid committed:
Slight tweaks to mobile view, updated modal

0xLienid committed:
Updating image sizing for mobile and tablet

0xJem committed:
WIP project info re-org

0xLienid committed:
Manage Donation modal re-titling

0xLienid committed:
Cause profile image size and info alignment

0xLienid committed:
Added new Your Donations paper

0xLienid committed:
Manage donation modal padding

0xLienid committed:
Abbreviate text to keep things on one line

0xJem committed:
Fix padding on CTA button

0xJem committed:
Adjust bottom padding for project details

0xJem committed:
Fix image sizing

0xJem committed:
Consistency in padding

0xJem committed:
Fix fill colour for the give sOHM icon

0xJem committed:
Fix issue with formatted string

0xJem committed:
Adjust spacing to meet spec

0xJem committed:
Adjust spacing according to spec

0xLienid committed:
Merge pull request #1379 from OlympusDAO/tyche-redeem

Tyche redeem

0xLienid committed:
Fixing Causes and Redeem padding

0xLienid committed:
cleaned up class naming for Redeem a bit

0xJem committed:
Manage donation modal respects breakpoints

0xLienid committed:
Fixed checking if wallet address is valid

0xLienid committed:
Merge branch 'tyche-causes' of github.com:OlympusDAO/olympus-frontend into tyche-causes
Pull from origin

0xLienid committed:
Remove console.logs

0xLienid committed:
Reset projects.json to have correct wallets

0xLienid committed:
Update cause profile image height

0xLienid committed:
Added useLocation to DepositRow to allow mock_sohm to work

0xLienid committed:
Moving as many things to one line as possible for My Donations

0xJem committed:
Don't spew "false" into element classes

0xJem committed:
TS refactor

0xJem committed:
Resolve lint warnings

0xJem committed:
Misc CSS cleanup

0xJem committed:
Adjust donations table to match spec

0xJem committed:
Adjust spacing of education card components (according to spec)

0xJem committed:
Correct padding on my donations screen

0xJem committed:
Merge branch 'develop' into tyche-causes

# Conflicts:
#	package.json
#	src/App.tsx
#	src/locales/translations
#	yarn.lock

0xJem committed:
Re-implement give routes. Also support URL re-writing when selecting tabs.

0xJem committed:
Simplify some code

0xJem committed:
Update snapshots with recent changes

0xLienid committed:
Light update to comments

0xJem committed:
Remove edit functionality from RecipientModal (shifted to ManageDonationModal)

0xJem committed:
Refactor RecipientModal to be easier to understand

0xJem committed:
Merge branch 'tyche-causes' of github.com:OlympusDAO/olympus-frontend into tyche-causes

0xLienid committed:
Better commenting on slice and helper logic

0xLienid committed:
Fixing non-handling of error on network changes

0xLienid committed:
Add grants cohort 0 info, images, and create new tab

0xLienid committed:
Render Grants cards and details pages

0xJem committed:
Shift buttons in RecipientModal to component-library

0xJem committed:
Project recipient field

0xJem committed:
Shift amount field in RecipientModal to the component library

0xJem committed:
Remove unused import

0xJem committed:
Shift recipient address to component librar

0xJem committed:
Replace SvgIcon

0xJem committed:
Merge branch 'develop' into tyche-causes

# Conflicts:
#	src/locales/translations
#	src/views/BondV2/Bond.scss
#	src/views/Give/Give.tsx
#	src/views/Give/RecipientModal.tsx
#	src/views/Give/__tests__/__snapshots__/Give.unit.test.tsx.snap

0xJem committed:
Fix alignment of the education graphics

0xJem committed:
RecipientModal: migrate to Modal from component-library

0xJem committed:
Update jest snapshot

0xJem committed:
Fix rendering bug with modal on Firefox

0xJem committed:
Shift to component-library's Modal

0xJem committed:
Cleanup

0xJem committed:
Shifted ManageDonationModal to be less complicated

0xJem committed:
Migrate buttons

0xJem committed:
Adjust spacing for ArrowGraphic

0xJem committed:
Migrate ManageDonationModal to component-library. Fix build warnings.

0xLienid committed:
Update images, change button sizing

0xJem committed:
Correct implementation of modal onClose and back button

0xJem committed:
Migrate RedeemYieldModal to component-library

0xJem committed:
Migrate RedeemYield to use component-library

0xJem committed:
Remove redundant view

0xJem committed:
Migrate ProjectCard to component-library

0xJem committed:
Migrate YieldRecipients to component-library

0xJem committed:
Migrate GiveInfo

0xJem committed:
Partial migration of Give to component-library

0xJem committed:
Migrate Give to component-library

0xJem committed:
Migrate DepositRow to component-library

0xJem committed:
Partial migration of Causes Dashboard to component-library

0xJem committed:
Workaround for component-library issue

0xJem committed:
Merge branch 'develop' into tyche-causes

0xJem committed:
Merge branch 'tyche-causes' into tyche-component-library

0xJem committed:
Update jest snapshots

0xJem committed:
Merge branch 'tyche-component-library' of github.com:OlympusDAO/olympus-frontend into tyche-component-library

0xJem committed:
Fix failing test

0xJem committed:
Remove redundant CSS after component-library migration

0xLienid committed:
Reworked styling, updated short description copy, and added grants explainer

0xJem committed:
Partial fixes for spacing in project details

0xJem committed:
Cleanup spacing in project details

0xJem committed:
Fix positioning of edit button

0xJem committed:
Adjust spacing for countdown timer

0xJem committed:
Fix right-alignment of sOHM Yield Sent

0xJem committed:
Merge branch 'develop' into tyche-component-library

brightiron committed:
move this one component back to muipaper

0xJem committed:
Merge pull request #1481 from OlympusDAO/brightiron/muipaper-tyche

move this one component back to muipaper

0xJem committed:
Remove obsolete Typography components, update to component-library 1.7.0

0xJem committed:
Merge branch 'tyche-component-library' of github.com:OlympusDAO/olympus-frontend into tyche-component-library

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-grants

0xJem committed:
Define schema for milestones and shift GrantsDashboard to use it

0xJem committed:
WIP milestones in grants card

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-component-library

0xJem committed:
Fix failing test due to changes

0xJem committed:
Merge branch 'tyche-component-library' into tyche-grants

0xJem committed:
Compilation fix

0xJem committed:
Shift GrantCard to component-library

0xJem committed:
WIP milestone progress indicator

0xJem committed:
Return to grants page

0xJem committed:
WIP implementation of the milestone progress bar (using an external library)

0xJem committed:
Complete spacing and styling of milestone progress bar, plus milestone details.

0xJem committed:
Handle out of bonds index

0xJem committed:
Milestone amount

0xJem committed:
WIP migration of the grant data section to a responsive Grid

0xJem committed:
Bold

0xLienid committed:
Working on icon updates

0xJem committed:
Complete layout of grant data

0xJem committed:
Merge branch 'tyche-grants' of github.com:OlympusDAO/olympus-frontend into tyche-grants

0xJem committed:
Capitalisation

0xJem committed:
Fix displaying donor count

0xJem committed:
Remove extraneous log

0xJem committed:
Fix button position on edit donation card

0xJem committed:
Shift Grant screen to use Grid instead of hacky CSS

0xJem committed:
Remove extraneous classes

0xJem committed:
About section layout

0xJem committed:
Fix positioning on grants profile

0xJem committed:
Correct sizing of grant image

0xJem committed:
Shift more elements to Grid

0xJem committed:
Shift donations section to Grid

0xJem committed:
Shift to using the topLeft prop of the Paper component and Grid

0xJem committed:
Fix sizing of grant image in medium breakpoint

0xJem committed:
Correct sizing of milestone details

0xJem committed:
Add new icons (though they have theme problems)

0xJem committed:
Correct sizing of SVG icons

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-grants

0xJem commented:
@appleseed-iii @mrunbanked I am waiting on design to make suggestions (e.g. spacing issues) and icons, but this is 99% done. If you could lend your eyes to this over the next few days, I'd appreciate it!

0xJem committed:
Update failing snapshot test

0xJem committed:
Remove redundant code

appleseed-iii commented:
> @appleseed-iii @mrunbanked I am waiting on design to make suggestions (e.g. spacing issues) and icons, but this is 99% done. If you could lend your eyes to this over the next few days, I'd appreciate it!

thank you for the heads up. Will review

0xJem committed:
Align grant profile with design 2.0.2

0xJem committed:
Add the ability to hide a grant from display

0xJem committed:
Update grants milestone amounts

0xJem committed:
Fix rendering of the donors icon with themes. This required explicitly adding a fill property, and defining the stroke property as currentcolor

0xJem committed:
Workaround for SVG fills

0xJem committed:
Add remaining 2 SVG icons for Grants (with stroke/fill fixes)

0xJem committed:
Fix wrapping of donors

0xJem committed:
Remove redundant grant

0xJem committed:
Didn't mean to enable that

0xJem committed:
Correct layout on md and lg breakpoints

0xJem committed:
Fix SVG rendering of fills

0xJem committed:
SVG updates
- Add new/updated icons
- Remove redundant icons
- Fix SVG background/fill issues

These SVG icons will be moved into component-library shortly.

0xJem committed:
Update EducationCard with latest SVG icons

0xJem committed:
Merge remote-tracking branch 'origin/develop' into tyche-grants

0xJem committed:
Hide project stats for grants

0xJem committed:
Merge remote-tracking branch 'origin/HEAD' into tyche-grants

0xJem committed:
Remove extraneous log

0xJem committed:
Update to component-library 1.7.1

0xJem committed:
Shift icons in GrantCard to component-library

0xJem committed:
Migrate ProjectCard to component-library icons

0xJem committed:
Shift EducationCard to use component-library icons

0xJem committed:
Remove redundant SVG files

0xJem committed:
Add feature flag to disable Give Grants

0xJem committed:
Misc design fixes

0xJem commented:
@mrunbanked @appleseed-iii I should hear back from design on the last few interface tweaks, but the codebase is 99% done. If you could give your feedback over the next few days, I'd appreciate it!

0xJem committed:
Update snapshots

0xJem committed:
Disable individual grant projects until they are approved. Also display helpful text if no grants are active.

0xJem committed:
Redundant Grid component

0xJem committed:
Remove redundant CSS

0xJem committed:
Shift CauseCard to use Grid component

0xJem committed:
WIP refactoring of ProjectCard to use Grid component

0xJem committed:
Remove redundant className

0xJem committed:
Remove redundancies

0xJem committed:
Revert temporary change to user donating flag

0xJem committed:
Remove redundant CSS

0xJem committed:
Remove redundant CSS

0xJem committed:
Update test snapshot with recent changes

0xJem committed:
Remove redundant CSS

0xJem committed:
Simplify the image used in Project and Grant screens

0xJem committed:
Convert amount screen of RecipientModal to use the Grid component

0xJem committed:
WIP conversion of RecipientModal to Grid component

0xJem committed:
Fix breakpoints on confirmation screen.

0xJem committed:
Align RecipientModal with reference design

0xJem committed:
Migrate GiveInfo. Remove redundant CSS.

0xJem committed:
Update snapshot

0xJem committed:
Shift CausesDashboard to use MUI components

0xJem committed:
Cleanup Give component

0xJem committed:
Update snapshot

0xJem committed:
Migrate CausesDashboard and GrantsDashboard

0xJem committed:
Clean up redundant CSS (LOTS of it)

0xJem committed:
Fix breakpoints for project and grant card

0xJem committed:
Responsive version of no donations message

0xJem committed:
Migrate manage donation screen

0xJem committed:
Migrate ManageDonationModal

0xJem committed:
Shift to use GiveBox

0xJem committed:
Convert RedeemYield to use Grid. Remove redundant CSS.

0xJem committed:
Use the DataRow component

0xJem committed:
Remove redundant component

0xJem committed:
Adjust breakpoint

0xJem committed:
Accidental commit

0xJem committed:
Migrate RedeemYieldModal to use Grid. Cleanup CSS.

0xJem committed:
Revert feature flag for give grants (since we will display an informative message instead)

0xJem committed:
Merge branch 'develop' into tyche-grants

# Conflicts:
#	.env.example
#	.env.test
#	src/locales/translations

0xJem committed:
Compilation fixes due to merge

0xJem committed:
Rebase fix

0xJem committed:
Layout fixes. Remove custom styling where possible.

0xJem committed:
Display message if there is no milestone information

0xJem committed:
Clean up EducationCard: shift to Grid components, remove redundant graphics, rename components.

0xJem committed:
Remove redundant SVG fill

0xJem commented:
Code and design is done. We are awaiting agreement signing for the grants (and that will involve minor edits to the JSON file), which can come in a later PR. Ready for review!

0xJem committed:
Remove hard-coded temporary value

0xJem committed:
Center-align the right-facing arrow

0xJem committed:
Backport text alignment

0xJem committed:
Update snapshot

0xJem committed:
Merge branch 'develop' into tyche-grants

0xJem committed:
Merge branch 'tyche-grants' into give-project-card

0xJem committed:
Merge pull request #1545 from OlympusDAO/give-project-card

Migrate Give screens to Grid component

0xJem committed:
Fix Paper width. Remove extraneous padding around causes/grant dashboards.

0xJem committed:
Apply fullWidth to Paper

0xJem committed:
Merge branch 'develop' into tyche-grants

# Conflicts:
#	src/views/Give/RecipientModal.tsx

0xJem committed:
Better handling of the main Give container at different breakpoints

0xJem committed:
Update snapshots

0xJem committed:
Enable thunks to record the grant name instead of a wallet address

0xJem committed:
Merge branch 'develop' into tyche-grants

# Conflicts:
#	src/App.tsx
#	src/components/__tests__/__snapshots__/EducationCard.unit.test.tsx.snap
#	src/views/Give/Give.scss
#	src/views/Give/__tests__/__snapshots__/Give.unit.test.tsx.snap
#	yarn.lock

0xJem committed:
Fix bug introduced in a recent version of component-library

appleseed-iii commented on src/views/Give/YieldRecipients.tsx:
```diff
@@ -48,65 +52,67 @@ export default function YieldRecipients({ changeView }: RecipientModalProps) {
-    </Grid>
+            <TableCell align="right">
+              <Typography variant="body1">
+                <Trans>YIELD SENT</Trans>
```
The alignment of the My Donations table is thrown off a bit on mobile? 
<img width="284" alt="image" src="https://user-images.githubusercontent.com/80423742/160191700-1fa32199-6d00-4ec4-80e1-a9713119de02.png">

appleseed-iii replied:
other than it looks good to go

appleseed-iii replied:
here's what it looks like on master right now 
<img width="289" alt="image" src="https://user-images.githubusercontent.com/80423742/160192416-80660b42-d062-473d-a037-b64564f03ab4.png">

0xJem replied:
Thanks for spotting this! YieldRecipients was the last section that hadn't been converted to use the Grid component (it was using a Table and I was wary of messing with that), but this problematic behaviour pushed me to make that conversion and fix this issue. (Which it has done.) It now matches the design spec.

Large breakpoint:
<img width="1025" alt="Screenshot 2022-03-28 at 11 58 26 AM" src="https://user-images.githubusercontent.com/91760036/160354195-7f98e42d-8909-43b8-a3dc-b07edfa4cae9.png">

Medium:
<img width="739" alt="Screenshot 2022-03-28 at 11 58 44 AM" src="https://user-images.githubusercontent.com/91760036/160354227-15f567d8-3d12-4ff3-b142-b07f8150116b.png">

Extra-small:
<img width="500" alt="Screenshot 2022-03-28 at 12 01 28 PM" src="https://user-images.githubusercontent.com/91760036/160354294-3d45726f-6ce1-4efe-bb64-84d410afb562.png">

Extra-small (mobile):
<img width="299" alt="Screenshot 2022-03-28 at 12 09 34 PM" src="https://user-images.githubusercontent.com/91760036/160354630-78fa0e81-037a-4e15-a942-ea7cb782fc44.png">

0xJem replied:
@appleseed-iii for your review

appleseed-iii replied:
thanks @0xJem . looks good

0xJem committed:
Back-port alignment fix from 8856ace4457429c398c2478760b30ba72329fd18

0xJem committed:
Migrate YieldRecipients to use Grid

0xJem committed:
Fix an issue with the selected tab being donations
